### DISCUSSION TITLE
Overloading local function with same name as enclosing function

### DISCUSSION THREAD

**Discussion by dotjpg3141** [dotnet/csharplang#633] [General]
Overloading local function with same name as the enclosing function gives a compiler error:

```cs
class Program
{
	static int Sum(int value)
	{
		int Sum()
		{
			return Sum(value - 1);
		}
		return value + Sum();
	}
}
```

**Expected behaviour:**
It compiles

**Actual behaviour:**
CS1501: No overload for method 'Sum' takes 1 arguments.

If I rename the local function to something other than `Sum` it will compile.

**Comment by svick** [comment]
I believe this is the expected behavior right now. Local functions don't behave like methods that can be overloaded, they behave like local variables. And if you have a local variable that has the same name as the enclosing method, only the local variable can be directly referenced.

If you're looking for a workaround, you could use `Program.Sum(value - 1)`.
